### PR TITLE
Allow adding items to empty worksheets

### LINKED ISSUE #1480: "Text" button cannot be launched in new, empty worksheets
**Describe the bug**
In a newly created worksheet, nothing happens when clicking "Text". 

**To Reproduce**
Steps to reproduce the behavior :
1. Create a new worksheet in the new front-end
2. Click on "Text"
3. Nothing happens

**Expected behavior**
Clicking "Text" would launch a new text box. 

**Screenshots**
http://g.recordit.co/MieNqsCTOC.gif

### PR DESCRIPTION
**Pull request by epicfaace**
Fixes #1480.

This is done by rendering an item on the frontend
with the `isDummyItem` attribute equal to true.

This PR also:
- Hides the text editor upon pressing the "Escape" key
- Adds an item to the end of the worksheet if no item is selected

image

### CODE REVIEW

File: frontend/src/components/worksheets/WorksheetItemList.js
Comment (percyliang) [inline frontend/src/components/worksheets/WorksheetItemList.js]: Make spacing consistent, also might be easier to read with multiple lines.

File: frontend/src/components/worksheets/items/ItemWrapper.js
Comment (percyliang) [inline frontend/src/components/worksheets/items/ItemWrapper.js:62]: these two variables don't seem like they type check

File: frontend/src/components/worksheets/items/ItemWrapper.js
Comment (percyliang) [inline frontend/src/components/worksheets/items/ItemWrapper.js:66]: what's the standard here - `null`?

File: frontend/src/components/worksheets/WorksheetItemList.js
Comment (percyliang) [inline frontend/src/components/worksheets/WorksheetItemList.js]: try to use `const` as you go

**Review by percyliang — CHANGES_REQUESTED** [review]
Why do we need a dummy element (would be nice to avoid special casing)?  You just need to be able to add a text block to an empty worksheet, no?

**Comment by epicfaace** [comment]
> Why do we need a dummy element (would be nice to avoid special casing)? You just need to be able to add a text block to an empty worksheet, no?

The text / run / upload blocks themselves are actually part of the `ItemWrapper` component -- so they can only show up _in relation to_ an existing item. That's why we need an existing `ItemWrapper` component (which stores dummy data) to get this to work.

File: frontend/src/components/worksheets/WorksheetItemList.js
Comment (epicfaace) [inline frontend/src/components/worksheets/WorksheetItemList.js]:
```suggestion
                    const focused = (index === this.props.focusIndex) ||
```

File: frontend/src/components/worksheets/items/ItemWrapper.js
Comment (epicfaace) [reply frontend/src/components/worksheets/items/ItemWrapper.js:66]: I don't think there's a standard -- either `null` or `""` work.

File: frontend/src/components/worksheets/items/ItemWrapper.js
Comment (epicfaace) [reply frontend/src/components/worksheets/items/ItemWrapper.js:62]: What do you mean? This is the same as saying

`const isDummyItem = item.isDummyItem;`

File: frontend/src/components/worksheets/WorksheetItemList.js
Comment (epicfaace) [inline frontend/src/components/worksheets/WorksheetItemList.js]:
```suggestion
            info.items = [
                {
                    "isDummyItem": true,
                    "text": "",
                    "mode": "markup_block",
                    "sort_keys": [
                        -1
                    ],
                    "ids": [
                        null
                    ],
                    "is_refined": true
                }
            ];
```

Comment (epicfaace) [reply frontend/src/components/worksheets/WorksheetItemList.js]: Updated, although for these things, I was thinking of adding `prettier` to our `lint-staged` step (https://github.com/codalab/codalab-worksheets/blob/frontend/frontend/package.json#L70). This would need to wait until the frontend is merged though so that the diff between master and frontend isn't too great.

File: frontend/src/components/worksheets/items/ItemWrapper.js
Comment (percyliang) [reply frontend/src/components/worksheets/items/ItemWrapper.js:62]: Oh, misread, never mind.

File: frontend/src/components/worksheets/WorksheetItemList.js
Comment (percyliang) [inline frontend/src/components/worksheets/WorksheetItemList.js:223]: `info.items.length > 0` should always hold now with the above.

**Comment by percyliang** [comment]
I'm still confused - we only want the upload/run/buttons at the header, not showing up between items (which was confusing, right?).  So I think we can get rid of ItemWrapper actually, no?

**Comment by epicfaace** [comment]
> I'm still confused - we only want the upload/run/buttons at the header, not showing up between items (which was confusing, right?). So I think we can get rid of ItemWrapper actually, no?

No, ItemWrapper doesn't show the actual upload/run/text buttons -- it shows the actual upload screen / run screen / text box screens.

**Pull request merged**
